File: OutboundFromElementPathChildrenCollection.cs

/*****************************************************************************
 *                                                                           *
 *  OutboundFromElementPathChildrenCollection.cs                             *
 *  25 November 2006                                                         *
 *  Project: Crystalline                                                     *
 *  Written by: Richard Sartor                                               *
 *  Copyright © 2006 Metaphysics Industries, Inc.                            *
 *                                                                           *
 *  Converted from C++ to C# on 10 November 2007                             *
 *                                                                           *
 *  A collection that manages the path connections outbound from an          *
 *    element.                                                               *
 *                                                                           *
 *****************************************************************************/

using System;
using System.Collections.Generic;
using MetaphysicsIndustries.Collections;

namespace MetaphysicsIndustries.Crystalline
{
    [Serializable]
    public class OutboundFromElementPathChildrenCollection : ElementPathChildrenCollection
	{
        private OutboundFromElementPathChildrenCollection() { }
		public OutboundFromElementPathChildrenCollection(Element parent)
            : base(parent)
		{
		}

        //public override void Dispose()
        //{
        //}

		protected override void SetParentElementOfPath(Path p, Element e)
		{
			p.From = e;
		}

		protected override Element GetParentElementOfPath(Path p)
		{
			return p.From;
		}

	}
}

File: Crystalline/OutboundFromElementPathChildrenCollection.cpp

/*****************************************************************************
 *                                                                           *
 *  OutboundFromElementPathChildrenCollection.cpp                            *
 *  25 November 2006                                                         *
 *  Project: Crystalline                                                     *
 *  Written by: Richard Sartor                                               *
 *  Copyright © 2006 Metaphysics Industries                                  *
 *                                                                           *
 *  A collection that manages the path connections outbound from an          *
 *    element.                                                               *
 *                                                                           *
 *****************************************************************************/



#include "stdafx.h"
#include "OutboundFromElementPathChildrenCollection.h"



;
NAMESPACE_START
;



OutboundFromElementPathChildrenCollection::OutboundFromElementPathChildrenCollection(Element^ e) : ElementPathChildrenCollection(e)
{
	
}

OutboundFromElementPathChildrenCollection::~OutboundFromElementPathChildrenCollection(void)
{
	
}

Element^ OutboundFromElementPathChildrenCollection::GetChildsParent(Path^ p)
{
	return p->From;
}

void OutboundFromElementPathChildrenCollection::SetChildsParent(Path^ p, Element^ e)
{
	p->From = e;
}



;
NAMESPACE_END
;




